tools/triage/add_triage_signature.py

```diff
@@ -2626,41 +2627,83 @@ def _process_ticket(self, url, issue_key):
             self._update_triaging_ticket(report)
 
 
-class MissingOSTreePivot(ErrorSignature):
+class OSTreeCommitMismatch(ErrorSignature):
     """
     This signature looks for missing pivot URL in rpm-ostree status of the control-plane hosts
     """
 
-    PIVOT_URL_PATTERN = re.compile(r"pivot://.*")
+    # machine_os_content tag name has been changing overtime, the first ones in
+    # the list have preference (since machine-os-content still exists in the
+    # release image)
+    machine_os_content_names = ["rhel-coreos", "rhel-coreos-8", "machine-os-content"]
+    # The name of the labels also have been changing
+    ostree_commit_labels = ["ostree.commit", "com.coreos.ostree-commit"]
+    # pull_secret = None
 
     def __init__(self, *args, **kwargs):
+        self.pull_secret = kwargs["pull_secret"]
         super().__init__(
             *args,
             **kwargs,
-            function_impact_label="missing_pivot_ostree",
-            comment_identifying_string="h1. MCO didn't pivot some hosts",
+            function_impact_label="ostree_commit_mismatch",
+            comment_identifying_string="h1. ostree commitid does not match the one in openshift release image",
         )
 
+    def get_os_content_image(self, release_image):
+        command = ["oc", "adm", "release", "info", release_image, "-ojson"]
+        if self.pull_secret:
+            command += ["--registry-config", self.pull_secret]
+        release_json = json.loads(subprocess.check_output(command))
+        tags = release_json["references"]["spec"]["tags"]
+        image_name = next(name for name in self.machine_os_content_names if name in [tag["name"] for tag in tags])
+        return next(tag["from"]["name"] for tag in tags if tag["name"] == image_name)
```

Comment:
Maybe a note here to say that the first tag matching one of the machine_os_content_names and the first name within that tag is OK as this is what we are expecting?

The iterator (next) syntax is quite hard to understand at first, I spent some time playing around and I now get it, but to the casual eye, this code is quite hard to read.

With that said, I like the compact nature.

I see that you have not specified a default for next() so will encounter a StopIteration error if none of the tags match, I think it would be an idea to handle this gracefully somehow so that you can tell the user that you could not find any of the machine_os_content names in tags.

Reply:
Agree on both

- code explained in a comment
- and controlled with a try catch when calling the function (I dont know if it's the best way to handle this ....)